### ISSUE TITLE
Investigate an Invalid string flatvector created

### ISSUE THREAD

**Issue by bikramSingh91** [facebookincubator/velox#6594]
### Description

Investigate an Invalid string flatvector created

### Error Reproduction

Artifacts here: https://app.circleci.com/pipelines/github/facebookincubator/velox/33535/workflows/f33a5f0d-ab24-4cec-879a-dd706fb4bc1e/jobs/217394/artifacts

### Relevant logs

```bash
I0914 22:41:09.986879  7236 ExpressionFuzzer.cpp:1313] ==============================> Started iteration 42770 (seed: 164314781)
I0914 22:41:09.992178  7236 ExpressionVerifier.cpp:86] Executing expression 0 : map_keys(map_from_entries(array_remove("c0",{17 elements starting at 97 {1974-01-26T00:28:22.243191064 => false, 2021-05-25T10:58:45.778943777 => false, 1977-09-29T15:03:14.477523984 => false, 2022-08-15T09:15:22.254280416 => false, 2015-04-04T09:04:25.257169497 => false, ...}, {224588267, [14->57] [57->22] [22->19] [19->74] 2018-09-27T07:13:00.104396282, 798917226, [14->37] [37->63] 0.5667001197580248, [14->47] ,;vP6D^M2)j}})))
I0914 22:41:10.011225  7236 FuzzerToolkit.cpp:103] Exceptions match.
I0914 22:41:10.011269  7236 ExpressionFuzzer.cpp:1353] Both paths failed with compatible exceptions. Retrying expression using try().
I0914 22:41:10.011312  7236 ExpressionVerifier.cpp:86] Executing expression 0 : try(map_keys(map_from_entries(array_remove("c0",{17 elements starting at 97 {1974-01-26T00:28:22.243191064 => false, 2021-05-25T10:58:45.778943777 => false, 1977-09-29T15:03:14.477523984 => false, 2022-08-15T09:15:22.254280416 => false, 2015-04-04T09:04:25.257169497 => false, ...}, {224588267, [14->57] [57->22] [22->19] [19->74] 2018-09-27T07:13:00.104396282, 798917226, [14->37] [37->63] 0.5667001197580248, [14->47] ,;vP6D^M2)j}}))))
I0914 22:41:10.032284  7236 ExpressionVerifier.cpp:74] All results match.
I0914 22:41:10.032480  7236 ExpressionVerifier.cpp:74] All results match.
I0914 22:41:10.032711  7236 ExpressionFuzzer.cpp:1274] Retrying original expression on 18 rows without errors
I0914 22:41:10.032748  7236 ExpressionVerifier.cpp:86] Executing expression 0 : map_keys(map_from_entries(array_remove("c0",{17 elements starting at 97 {1974-01-26T00:28:22.243191064 => false, 2021-05-25T10:58:45.778943777 => false, 1977-09-29T15:03:14.477523984 => false, 2022-08-15T09:15:22.254280416 => false, 2015-04-04T09:04:25.257169497 => false, ...}, {224588267, [14->57] [57->22] [22->19] [19->74] 2018-09-27T07:13:00.104396282, 798917226, [14->37] [37->63] 0.5667001197580248, [14->47] ,;vP6D^M2)j}})))
E0914 22:41:10.033993  7236 Exceptions.h:69] Line: ../../velox/vector/FlatVector.cpp:530, Function:validate, Expression: isValid String view at idx 12 points outside of the string buffers, Source: RUNTIME, ErrorCode: INVALID_STATE
E0914 22:41:10.034715  7236 ExpressionVerifier.cpp:176] Common eval: Exceptions other than VeloxUserError are not allowed.
I0914 22:41:10.035738  7236 ExpressionVerifier.cpp:346] Persisted input: --fuzzer_repro_path /tmp/fuzzer_repro/velox_expressionVerifier_EljFCz --input_path /tmp/fuzzer_repro/velox_expressionVerifier_EljFCz/input_vector --result_path /tmp/fuzzer_repro/velox_expressionVerifier_EljFCz/result_vector --sql_path /tmp/fuzzer_repro/velox_expressionVerifier_EljFCz/sql --lazy_column_list_path /tmp/fuzzer_repro/velox_expressionVerifier_EljFCz/indices_of_lazy_columns --complex_constant_path /tmp/fuzzer_repro/velox_expressionVerifier_EljFCz/complex_constants
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: String view at idx 12 points outside of the string buffers
Retriable: False
Expression: isValid
Context: map_from_entries(array_remove(c0, {17 elements starting at 0 {1974-01-26T00:28:22.243191064 => false, 2021-05-25T10:58:45.778943777 => false, 1977-09-29T15:03:14.477523984 => false, 2022-08-15T09:15:22.254280416 => false, 2015-04-04T09:04:25.257169497 => false, ...}, {224588267, 2018-09-27T07:13:00.104396282, 798917226, 0.5667001197580248, ,;vP6D^M2)j}}:ROW<"":MAP<TIMESTAMP,BOOLEAN>,"":ROW<f0:DATE,f1:TIMESTAMP,f2:DATE,f3:DOUBLE,f4:VARCHAR>>))
Top-Level Context: map_keys(map_from_entries(array_remove(c0, {17 elements starting at 0 {1974-01-26T00:28:22.243191064 => false, 2021-05-25T10:58:45.778943777 => false, 1977-09-29T15:03:14.477523984 => false, 2022-08-15T09:15:22.254280416 => false, 2015-04-04T09:04:25.257169497 => false, ...}, {224588267, 2018-09-27T07:13:00.104396282, 798917226, 0.5667001197580248, ,;vP6D^M2)j}}:ROW<"":MAP<TIMESTAMP,BOOLEAN>,"":ROW<f0:DATE,f1:TIMESTAMP,f2:DATE,f3:DOUBLE,f4:VARCHAR>>)))
Function: validate
File: ../../velox/vector/FlatVector.cpp
Line: 530
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxException5State4makeIZNS1_C4EPKcmS5_St17basic_string_viewIcSt11char_traitsIcEES9_S9_S9_bNS1_4TypeES9_EUlRT_E_EESt10shared_ptrIKS2_ESA_SB_
# 2  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 3  _ZN8facebook5velox17VeloxRuntimeErrorC2EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bS7_
# 4  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
# 5  _ZNK8facebook5velox10FlatVectorINS0_10StringViewEE8validateERKNS0_21VectorValidateOptionsE
# 6  _ZNK8facebook5velox9RowVector8validateERKNS0_21VectorValidateOptionsE
# 7  _ZNK8facebook5velox9MapVector8validateERKNS0_21VectorValidateOptionsE
# 8  _ZN8facebook5velox4exec12_GLOBAL__N_124checkResultInternalStateERSt10shared_ptrINS0_10BaseVectorEE
# 9  _ZN8facebook5velox4exec4Expr4evalERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEEPKNS1_7ExprSetE
# 10 _ZZN8facebook5velox4exec4Expr11evalAllImplERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEEENKUlT_E_clIiEEDaSC_
# 11 _ZN8facebook5velox4exec4Expr20evalArgsDefaultNullsIZNS2_11evalAllImplERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEEEUlT_E_EEbRNS1_20MutableRemainingRowsESD_S8_SC_
# 12 _ZN8facebook5velox4exec4Expr11evalAllImplERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEE
# 13 _ZN8facebook5velox4exec4Expr7evalAllERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEE
# 14 _ZN8facebook5velox4exec4Expr13evalWithNullsERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEE
# 15 _ZN8facebook5velox4exec4Expr12evalWithMemoERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEE
# 16 _ZN8facebook5velox4exec4Expr13evalEncodingsERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEE
# 17 _ZN8facebook5velox4exec4Expr4evalERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt10shared_ptrINS0_10BaseVectorEEPKNS1_7ExprSetE
# 18 _ZN8facebook5velox4exec7ExprSet4evalEiibRKNS0_17SelectivityVectorERNS1_7EvalCtxERSt6vectorISt10shared_ptrINS0_10BaseVectorEESaISB_EE
# 19 _ZN8facebook5velox4exec7ExprSet4evalERKNS0_17SelectivityVectorERNS1_7EvalCtxERSt6vectorISt10shared_ptrINS0_10BaseVectorEESaISB_EE
# 20 _ZN8facebook5velox4test18ExpressionVerifier6verifyERKSt6vectorISt10shared_ptrIKNS0_4core10ITypedExprEESaIS8_EERKS4_INS0_9RowVectorEEOS4_INS0_10BaseVectorEEbS3_IiSaIiEE
# 21 _ZN8facebook5velox4test16ExpressionFuzzer12retryWithTryESt6vectorISt10shared_ptrIKNS0_4core10ITypedExprEESaIS8_EERKS4_INS0_9RowVectorEERKS4_INS0_10BaseVectorEERKS3_IiSaIiEE
# 22 _ZN8facebook5velox4test16ExpressionFuzzer2goEv
# 23 _ZN8facebook5velox4test16expressionFuzzerESt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt6vectorIPKNS0_4exec17FunctionSignatureESaISD_EESt4hashIS8_ESt8equal_toIS8_ESaISt4pairIKS8_SF_EEEm
# 24 _ZN12FuzzerRunner12runFromGtestERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEmRKSt13unordered_setIS5_St4hashIS5_ESt8equal_toIS5_ESaIS5_EES7_
# 25 _ZN12FuzzerRunner3runERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEmRKSt13unordered_setIS5_St4hashIS5_ESt8equal_toIS5_ESaIS5_EES7_
# 26 main
# 27 __libc_start_main
# 28 _start
```

**Comment by bikramSingh91** [comment]
I tried to repro this both by using the artifacts and by using the seed in the fuzzer. I am however not able to repro this. I combed through the code to figure out what i might be missing. My best guess (a pretty wild guess) is that this might be due to the re-use of a string flat vector from the vector pool which did not get properly resized and has garbage value. I will run the fuzzer for a longer time (30 mins) to see if i can replicate this. But i'll have to get creative in figuring out how to debug this. Keeping this issue open and will get back to it as time permits.
